src/askem_beaker/contexts/dataset/procedures/python3/hmi_create_csv_dataset.py

```diff
@@ -0,0 +1,39 @@
+import os
+import requests
+import tempfile
+
+from json import JSONDecodeError
+
+with tempfile.TemporaryFile() as temp_csv_file:
+    {{ var_name|default("df") }}.to_csv(temp_csv_file, index=False, header=True)
+    temp_csv_file.seek(0)
+    # Get the HMI_SERVER endpoint and auth token from the environment variable
+    hmi_server = os.getenv('HMI_SERVER_URL')
```

Comment:
Procedures aren't guaranteed to be run on the same machine as the the beaker kernel. It is advised to fetch the env variable in the context.py file and pass that in as a template variable instead.

Reply:
fixed